### PR TITLE
Limit access to secrets to the admins only

### PR DESCRIPTION
**Pull request by smahmed776**

### CODE REVIEW

File: src/renderer/lib/base-client.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 import axios from "axios";
 
-export const baseClient = axios.create({
-  baseURL: API_URI,
-  withCredentials: true,
-  headers: {
-    "Content-Type": "application/json",
-  },
-});
+export const baseClient = (basicAuth?: `${string}:${string}`) =>
```
Comment (itsjoeoui) [inline src/renderer/lib/base-client.ts:4]: when you do `withCredentials: true`, credentials such as cookies, auth headers, etc will already be included with the request, assume the cookie is configured properly you do not need to attach it again like this. Now also every time you are initializing a new client upon request which is not efficient

**Review by itsjoeoui — CHANGES_REQUESTED** [review]
lgtm

File: src/renderer/lib/base-client.ts (hunk 1 above)
Comment (smahmed776) [reply src/renderer/lib/base-client.ts]: Thanks, I've adopted cookie-based auth system for backend.

**Review by itsjoeoui — APPROVED** [review]
lgtm

**Pull request merged**